src/InputBase.js

```diff
@@ -91,17 +92,20 @@ class InputBase extends React.Component {
   }
 
   renderIcon() {
-    let classes = {
-      'glyphicon': true,
-      'form-control-feedback': true,
-      'glyphicon-ok': this.props.bsStyle === 'success',
-      'glyphicon-warning-sign': this.props.bsStyle === 'warning',
-      'glyphicon-remove': this.props.bsStyle === 'error'
-    };
+    if (this.props.hasFeedback) {
+      if (this.props.feedbackIcon) {
+        return this.props.feedbackIcon;
```

Comment:
nope, that was the point of adding the prop, it'll just work in the 90% case of a user just using a Gylphicon, but if they want to provide some custom icon component that handles form feeback different'y then can provide one